src/vconsole/vconsole-setup.c

```diff
@@ -610,14 +621,31 @@ static int run(int argc, char **argv) {
 
         umask(0022);
 
-        if (argv[1])
+        if (argv[1]) {
                 fd = verify_source_vc(&vc, argv[1]);
-        else
+                if (fd < 0)
+                        return EXIT_FAILURE;
```

Comment:
btw, just do `return fd` here... the main-func.h boilerplate turns that into EXIT_FAILURE anyway